unittests/FEXLinuxTests/tests/signal/invalid_util.h

```diff
@@ -24,3 +25,159 @@ static void CapturingHandler(int signal, siginfo_t *siginfo, void* context) {
   _context->uc_mcontext.gregs[FEX_IP_REG] += capturing_handler_skip;
 #undef FEX_IP_REG
 }
+
+#if __SIZEOF_POINTER__ == 4
+struct sigcontext_32 {
+  uint16_t gs, gsh;
+  uint16_t fs, fsh;
+  uint16_t es, esh;
+  uint16_t ds, dsh;
+  uint32_t di;
+  uint32_t si;
+  uint32_t bp;
+  uint32_t sp;
+  uint32_t bx;
+  uint32_t dx;
+  uint32_t cx;
+  uint32_t ax;
+  uint32_t trapno;
+  uint32_t err;
+  uint32_t ip;
+  uint16_t cs, csh;
+  uint32_t flags;
+  uint32_t sp_at_signal;
+  uint16_t ss, ssh;
+
+  uint32_t fpstate;
+  uint32_t oldmask;
+  uint32_t cr2;
+};
+struct sigframe_ia32 {
+  uint32_t pretcode;
+  int signal;
+  sigcontext_32 sc;
+  // <...>
+  // Some extra state
+};
+
+struct rt_sigframe_ia32 {
+  uint32_t pretcode;
+  int signal;
+  uint32_t pinfo;
+  uint32_t puc;
+  siginfo_t info;
+  ucontext_t uc;
+  // <...>
+  // Some extra state
+};
+
+struct CapturedHandlerState_32 {
+  sigcontext_32 mctx;
+  int signal;
+  int si_code;
+};
+
+struct CapturedHandlerState_regparm_32 {
+  int signal;
+  siginfo_t *siginfo;
+  void* context;
+};
+
+
+
+std::optional<CapturedHandlerState_32> from_handler_32;
+std::optional<CapturedHandlerState_regparm_32> from_handler_regparm_32;
+
+static void CapturingHandler_non_sainfo(int signal, ...) {
```

Comment:
Could you add docstrings for the new functions and variables? Then for future tests, people can decide which one is appropriate without deeper knowledge of signal handling differences.

Reply:
Added comments.